patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -454,7 +454,7 @@
 +     * Queries the percentage of the 'Durability' bar that should be drawn.
 +     *
 +     * @param stack The current ItemStack
-+     * @return 1.0 for 100% 0 for 0%
```

Comment:
the default impl is stack.damage / stack.maxdamage 

if stack.damage = 1 the method returns something low causing the bar to draw fully.
if stack.damage = maxdamage the method returns 1 which causes the bar to draw empty.